sktime/transformations/series/boxcox.py

```diff
@@ -111,69 +115,77 @@ def __init__(self, bounds=None, method="mle", sp=None):
         self.sp = sp
         super(BoxCoxTransformer, self).__init__()
 
-    def _fit(self, Z, X=None):
-        """Fit data.
+    def _fit(self, X, y=None):
+        """
+        Fit transformer to X and y.
+
+        core logic
 
         Parameters
         ----------
-        Z : pd.Series
-            Series to fit.
-        X : pd.DataFrame, optional (default=None)
-            Exogenous data used in transformation.
+        X : 2D np.ndarray
+            Data to be transformed
+        y : ignored argument for interface compatibility
+            Additional data, e.g., labels for transformation
 
         Returns
         -------
-        self
+        self: a fitted instance of the estimator
         """
-        z = check_series(Z, enforce_univariate=True)
+        X = X.flatten()
         if self.method != "guerrero":
-            self.lambda_ = _boxcox_normmax(z, bounds=self.bounds, method=self.method)
+            self.lambda_ = _boxcox_normmax(X, bounds=self.bounds, method=self.method)
         else:
-            self.lambda_ = _guerrero(z, self.sp, self.bounds)
+            self.lambda_ = _guerrero(X, self.sp, self.bounds)
 
         return self
 
-    def _transform(self, Z, X=None):
-        """Transform data.
+    def _transform(self, X, y=None):
+        """Transform X and return a transformed version.
+
+        core logic
 
         Parameters
         ----------
-        Z : pd.Series
-            Series to transform.
-        X : pd.DataFrame, optional (default=None)
-            Exogenous data used in transformation.
+        X : 2D np.ndarray
+            Data to be transformed
+        y : ignored argument for interface compatibility
+            Additional data, e.g., labels for transformation
 
         Returns
         -------
-        Zt : pd.Series
-            Transformed series.
+        Xt : 2D np.ndarray
+            transformed version of X
         """
-        z = check_series(Z, enforce_univariate=True)
-        zt = boxcox(z.to_numpy(), self.lambda_)
-        return pd.Series(zt, index=z.index)
+        X_shape = X.shape
+        Xt = boxcox(X.flatten(), self.lambda_)
```

Comment:
hm, yes, you're right.
I'm not sure anymore why I changed this.

Question: would it make more sense to do this by variable then? Or should users apply the `ColumnTransformer` instead?
Question: would it make sense later to have a transformer where `lambda` can be input?

Reply:
I've now set the `univariate-only` tag to `True`. That should allow only n x 1 arrays (i.e., univariate series in the numpy format), and all should be fine?

Reply:
Thanks! yes this should work fine as now the length will be addressed as intended 👍 
Q1: I think it would be more user-friendly if we add functionality to handle multivariate input here, either by calling ColumnTransformer from BoxCox or adding the variable-wise transform. Don't know how far we'd like to go with complicating it though, in terms of design it seems to be a bit excessive. My vote would be that even though it's easier for a user to have package deals, we should leave it modular.
Q2: It depends whether there's a scenario this transformation will be used for some other reasons rather than approximating the data to normal distribution. I don't think it's necessary to enable input here, but maybe you have some use cases in mind?
An enhancement that we can add here might be adding [Yeo-Johnson method](https://www.jstor.org/stable/2673623), enabling negative values. What do you think?

Reply:
Q1: yes, I've been thinking about that. Currently, my favourite solution would be "apply-by-variable" next to "apply-by-instance", i.e., univariate forecasters/transformers are cast across variables, just like simple forecasters are cast across instances in a panel. That may require some additional work, and in most cases a direct implemention inside the transformer would be more efficient.

Q2: hm, not sure - generally, I like to not make too many assumptions on how things will or have to be used, since often there are perfectly sensible application cases outside of what one can think of individually. Regarding Yeo-Johnson, this sounds like a sensible extension? Would allow to cover negative values, and up to the strategy of how lambda is determined, it is the same as BoxCox in the case we currently cover (all values positive)?